lxd/archive/archive.go

```diff
@@ -0,0 +1,71 @@
+package archive
+
+import (
+	"archive/tar"
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/lxc/lxd/lxd/apparmor"
+	"github.com/lxc/lxd/lxd/state"
+	"github.com/lxc/lxd/shared/subprocess"
+)
+
+func ExtractWithFds(name string, args []string, stdin io.Reader, stdout io.Writer, s *state.State, profileName string) error {
+	apparmor.ArchiveLoad(s, profileName)
+	defer apparmor.ArchiveUnload(s, profileName)
+
+	p, err := subprocess.NewProcessWithFds(name, args, stdin, stdout)
+	if err != nil {
+		return fmt.Errorf("Failed to start unpack: Failed to creating subprocess: %w", err)
+	}
+
+	p.SetApparmor(apparmor.ArchiveProfileName(profileName))
+
+	err = p.Start()
+	if err != nil {
+		return fmt.Errorf("Failed to start unpack: Failed running: tar: %w", err)
+	}
+
+	p.Wait(context.Background())
+	return nil
+}
+
+func CompressedTarReader(ctx context.Context, r io.ReadSeeker, unpacker []string, s *state.State, profileName string) (*tar.Reader, context.CancelFunc, error) {
```

Comment:
Please can you add some function description comments to these functions, for example from an initial read its not clear what the `unpack []string` argument is for/takes.